File: LeetCode/src/LeetCode11.java

//Container With the Most Water
public class LeetCode11 {
    public int maxArea(int[] height) {
        int first = 0;
        int last = height.length - 1;
        int maxArea = 0;

        while (first < last) {
            if (height[first] < height[last]) {
                maxArea = Math.max(maxArea, height[first] * (last - first));
                first++;
            } else {
                maxArea = Math.max(maxArea, height[last] * (last - first));
                last--;
            }
        }

        return maxArea;
    }
}

File: LeetCode/src/LeetCode11.py
from typing import List

#2 pointer appraoch for containter with the most water
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxW = 1
        x = 0
        y = len(height)-1
        while x<y:
            if height[x] <= height[y]:
                maxW = max(maxW, height[x]*(y-x))
                x+=1
            elif height[y] <= height[x]:
                maxW = max(maxW, height[y]*(y-x))
                y-=1
        return maxW

x=Solution()
height = [1,1]

print(x.maxArea(height))